### PR TITLE
Allow to use a specific branch of composer

### PR DESCRIPTION
**Pull request by eexit**
Hello,

[Composer 2 has released](https://blog.packagist.com/composer-2-0-is-now-available/) and I think it would be nice to allow us to pick which major version to use.

Feel free to suggest a better name for the variable.

Thanks,

### CODE REVIEW

**Comment by eexit** [comment]
Working example log:

```
amazon-ebs: TASK [composer : Run Composer installer.] **************************************
    amazon-ebs: changed: [127.0.0.1] => {"changed": true, "cmd": ["php", "composer-installer.php", "--1"], "delta": "0:00:00.424894", "end": "2020-10-28 16:50:01.429910", "rc": 0, "start": "2020-10-28 16:50:01.005016", "stderr": "", "stderr_lines": [], "stdout": "All settings correct for using Composer\nDownloading...\n\nComposer (version 1.10.16) successfully installed to: /tmp/composer.phar\nUse it: php composer.phar", "stdout_lines": ["All settings correct for using Composer", "Downloading...", "", "Composer (version 1.10.16) successfully installed to: /tmp/composer.phar", "Use it: php composer.phar"]}
```

File: defaults/main.yml [hunk 1]
```diff
@@ -2,6 +2,7 @@
 composer_path: /usr/local/bin/composer
 composer_keep_updated: false
 composer_version: ''
+composer_version_branch: '--1'
```
Comment (woemar) [inline defaults/main.yml:5]: I would prefer to see '--2' as default value even is this means you have to adopt your config for older projects.

Comment (eexit) [reply defaults/main.yml]: Hi,

Done.
Beware though because Composer 2 is still very new and using it right now might be breaking plugins.

**Comment by eexit** [comment]
Hey @geerlingguy,

Would you mind taking a look at this PR when you get a chance? The CI has failed because Galaxy was down. Could you re-trigger it?

Thanks.

**Comment by geerlingguy** [comment]
@eexit - Can you rebase so it gets tested on GitHub Actions instead?

**Comment by eexit** [comment]
@geerlingguy Done!

**Comment by Lusitaniae** [comment]
Should we keep v1 as default?

I'm using this role to manage drupal websites and builds started failing already. 

```
  Problem 1
    - cweagans/composer-patches is locked to version 1.6.7 and an update of this package was not requested.
    - cweagans/composer-patches 1.6.7 requires composer-plugin-api ^1.0 -> found composer-plugin-api[2.0.0] but it does not match the constraint.

```
There's more errors after this.

Of course I can update the playbooks to use v1 and/or upgrade packages to a newer version that supports Composer 2 (if all of them do already), but a safe non breaking default would certainly be helpful.

**Comment by woemar** [comment]
I believe using composer 1 as default is not a good solution, as the composer role should always provide direct access to the current composer version.
Many older projects will need some modification of their playbook but that's a really small price to pay as the changes are minimal.

**Comment by Lusitaniae** [comment]
Should it have been a new major release (2.0.0) instead of 1.9.0 in that case?

**Comment by geerlingguy** [comment]
@Lusitaniae - Note that all the older releases of this role (prior to 1.9.0) would have also installed v2 by default at this point.

It might've been nice to do a 2.0.0 anyways, though, just to keep people a little more on their toes with regard to changes, but at this point, I would highly recommend either locking in v1 in your playbook vars or upgrading all your dependencies to work with V2. I know most people in the Composer dev community are already making plans for when v1 will be deprecated, and I'm guessing they won't wait indefinitely. V2 is night-and-day better, especially for speed an memory usage (I can deploy things to 1 GB VPSes again, yay!).

**Comment by rsicart** [comment]
This PR is a breaking change for all people using a fixed version on major version 1.

If both `composer_version_branch` (by default `--2` now) and `composer_version` are defined, only `composer_version_branch` is used.

 As a workaround for someone who needs to still use a fixed version of major version 1, you need to configure those 2 vars:

```
composer_version_branch: ''
composer_version: '1.10.12'
```

But the best option is to upgrade to major version 2.

**Comment by geerlingguy** [comment]
@rsicart - Ah, good point. I think it would be good to document this in the README. Would you or someone else like to put a new section in the variables like:

```
### Staying on Composer 1

While projects are upgrading to support Composer 2, it may be helpful to run Composer 1 instead. To do that, set these variables:

\`\`\`
composer_version_branch: ''
composer_version: '1.10.12'
\`\`\`
```

**Comment by eexit** [comment]
@rsicart @geerlingguy

Staying on version 1 of composer is simpler:

```
composer_version_branch: '--1'
```

I never used `composer_version` because it locks to a specific version of composer which does not make much sense unless your project is VERY dependent on composer internal changes. I don't see the point to lock in a specific version of composer when it comes to minor or patch.

**Comment by woemar** [comment]
May be obvious but the version number of this role should respect semantic versioning and change from 1.x to 2.x to reflect the fact that composer 2 is now the default because this may be a breaking change for some/most projects.
So people dependent on their composer 1 configuration may stick to 1.x while all other people may update to 2.x.

**Comment by rsicart** [comment]
@eexit On real life infrastructures, fixing versions is a best practice: we want to have equal staging and production environments.
Latest is not a version, using latest breakes idempotence.

**Pull request merged**